ext/standard/pack.c

```diff
@@ -62,40 +74,19 @@ typedef ZEND_SET_ALIGNED(1, uint64_t unaligned_uint64_t);
 typedef ZEND_SET_ALIGNED(1, unsigned int unaligned_uint);
 typedef ZEND_SET_ALIGNED(1, int unaligned_int);
 
-/* Mapping of byte from char (8bit) to long for machine endian */
-static int byte_map[1];
-
-/* Mappings of bytes from int (machine dependent) to int for machine endian */
-static int int_map[sizeof(int)];
-
-/* Mappings of bytes from shorts (16bit) for all endian environments */
-static int machine_endian_short_map[2];
-static int big_endian_short_map[2];
-static int little_endian_short_map[2];
-
-/* Mappings of bytes from longs (32bit) for all endian environments */
-static int machine_endian_long_map[4];
-static int big_endian_long_map[4];
-static int little_endian_long_map[4];
-
-#if SIZEOF_ZEND_LONG > 4
-/* Mappings of bytes from quads (64bit) for all endian environments */
-static int machine_endian_longlong_map[8];
-static int big_endian_longlong_map[8];
-static int little_endian_longlong_map[8];
-#endif
-
 /* {{{ php_pack */
-static void php_pack(zval *val, size_t size, int *map, char *output)
+static void php_pack(const zval *val, size_t size, php_pack_endianness endianness, char *output)
 {
-	size_t i;
-	char *v;
+	zend_ulong zl = zval_get_long(val);
 
-	convert_to_long(val);
-	v = (char *) &Z_LVAL_P(val);
+	const char *v = (const char *) &zl;
+	if ((endianness == PHP_LITTLE_ENDIAN) != MACHINE_LITTLE_ENDIAN) {
+		zl = PHP_LONG_BSWAP(zl);
+		v += sizeof(zend_long) - size;
```

Comment:
Should this either be one of those to be extra correct with types?
```suggestion
		v += sizeof(zl) - size;
```
```suggestion
		v += sizeof(zend_ulong) - size;
```